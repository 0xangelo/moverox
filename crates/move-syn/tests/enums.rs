use move_syn::File;
use testresult::TestResult;
use unsynn::{IParse as _, ToTokens as _};

macro_rules! include_move {
    ($path: literal) => {
        include_str!(concat!("../../../move/", $path))
    };
}

#[test]
fn main_() -> TestResult {
    let ast: File = include_move!("enums/sources/main.move")
        .to_token_iter()
        .parse_all()?;
    insta::assert_snapshot!(ast.tokens_to_string());
    Ok(())
}
