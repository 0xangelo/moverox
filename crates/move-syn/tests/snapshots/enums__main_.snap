---
source: crates/move-syn/tests/enums.rs
expression: ast.tokens_to_string()
---
module enums :: enums ; use std :: string :: String ; public enum Single { Only , } # [doc = " `Segment` enum definition."] # [doc = " Defines various string segments."] public enum Segment has copy , drop { # [doc = " Empty variant, no value."] Empty , # [doc = " Variant with a value (positional style)."] String (String) , # [doc = " Variant with named fields."] Special { content : vector < u8 > , encoding : u8 , } , } public enum Generic < phantom T > has copy , drop , store { Unit , Tuple (u64) , Struct { value : u64 , } }
